### PR TITLE
Added gh workflow trigger task and a job in run-test pipeline

### PR DESCRIPTION
**Pull request by jaydee029**
For issue
- galasa-dev/projectmanagement#1934

### CODE REVIEW

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Review by jadecarino — CHANGES_REQUESTED** [review]
Hi Dhruv - A few minor changes to make please, but really good work for your first time creating something for Tekton. I'm hoping it gave you a good insight into the differences between it and GH Actions. A couple changes for clarity and then 2 minor fixes.

File: pipelines/tasks/gh-workflow-trigger.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          key: password
+          optional: false
+    # onError: continue
+    script: |
+      #!/bin/sh
+      set +e
+
+     echo $GH_TOKEN | gh auth login --with-token  # not necessary just incase authentication fails 
+     gh workflow run $(params.workflowName) --repo https://github.com/galasa-dev/$(params.repoName) --ref $(params.refBranch)
+
```
[the remark this answers is not in the document]
Comment (jaydee029) [reply pipelines/tasks/gh-workflow-trigger.yaml:41]: done

File: pipelines/tasks/gh-workflow-trigger.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        secretKeyRef:
+          name: GALASA_TEAM_GITHUB_TOKEN
+          key: password
+          optional: false
+    # onError: continue
+    script: |
+      #!/bin/sh
+      set +e
+
+     echo $GH_TOKEN | gh auth login --with-token  # not necessary just incase authentication fails
```
[the remark this answers is not in the document]
Comment (jaydee029) [reply pipelines/tasks/gh-workflow-trigger.yaml]: donr

File: pipelines/tasks/gh-workflow-trigger.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    type: string    
+  steps:
+  - name: trigger-workflow
+    image: ghcr.io/galasa-dev/gitcli:main
+    imagePullPolicy: IfNotPresent
+    env:
+    - name: GH_TOKEN
+      valueFrom:
+        secretKeyRef:
+          name: GALASA_TEAM_GITHUB_TOKEN
```
[the remark this answers is not in the document]
Comment (jaydee029) [reply pipelines/tasks/gh-workflow-trigger.yaml]: done

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

File: pipelines/tasks/gh-workflow-trigger.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          key: password
+          optional: false
+    # onError: continue
+    script: |
+      #!/bin/sh
+      set +e
+
+     echo $GH_TOKEN | gh auth login --with-token  # not necessary just incase authentication fails 
+     gh workflow run $(params.workflowName) --repo https://github.com/galasa-dev/$(params.repoName) --ref $(params.refBranch)
```
[the remark this answers is not in the document]
Comment (jaydee029) [reply pipelines/tasks/gh-workflow-trigger.yaml]: done

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**